code/datums/martial/monk.dm

```diff
@@ -0,0 +1,455 @@
+		START_PROCESSING(SSfastprocess, src)
+		H.hairstyle = "Bald"
+		H.update_body()
+		H.update_hair()
+
+/datum/martial_art/monk/proc/flurry_penalty()
+	if(using_flurry)
+		return flurry_of_blows_penalty
+	return 0
+
+/datum/martial_art/monk/on_remove(mob/living/carbon/human/H)
+	to_chat(H, "<span class = 'userdanger'>You forget the ways of a Monk...</span>")
+	stunning_fist.Remove(H)
+	quivering_palm.Remove(H)
+	flurry_toggle.Remove(H)
+	monk_rest.Remove(H)
+	wholeness_of_body.Remove(H)
+	make_disciple.Remove(H)
+	REMOVE_TRAIT(H, TRAIT_NOGUNS, MONK_TRAIT)
+	STOP_PROCESSING(SSfastprocess, src)
+
+/datum/martial_art/monk/process()
+	..()
+	if(diamond_body)
+		owner.reagents.remove_all_type(/datum/reagent/toxin, 30)
+
+/datum/martial_art/monk/proc/do_level_up()
+	if(current_level % 4 == 0)
+		curr_ab++
+	switch(current_level)
+		if(2)
+			to_chat(owner, "<span class = 'danger'>You have practiced the Stunning Fist technique enough to use it in combat.</span>")
+			stunning_fist.Grant(owner)
+			stunning_fist.martial = src
+			available_actions = list(stunning_fist)
+		if(4)
+			to_chat(owner, "<span class = 'danger'>Your Ki has begun to develop, probing at the armor of your foes.</span>")
+			ki_level = 50
+		if(5)
+			to_chat(owner, "<span class = 'danger'>You feel more confident in your strikes, and your body is pure.</span>")
+			flurry_of_blows_penalty = -1
+			purity_of_body = TRUE
+		if(7)
+			to_chat(owner, "<span class = 'danger'>You can perform more Stunning Fists before resting. You have figured out the wholeness of your body.</span>")
+			stunning_fist.max_uses = 7
+			wholeness_of_body.Grant(owner)
+			wholeness_of_body.martial = src
+			available_actions = list(stunning_fist, wholeness_of_body)
+		if(8)
+			to_chat(owner, "<span class = 'danger'>You move from one downed target to the next quickly.</span>")
+			cleave_level = 1
+		if(9)
+			to_chat(owner, "<span class = 'danger'>You have become completely confident in your ability to land a strike.</span>")
+			flurry_of_blows_penalty = 0
+		if(10)
+			to_chat(owner, "<span class = 'danger'>Your Ki helps you seek the holes in enemy armor.</span>")
+			ki_level = 75
+		if(11)
+			to_chat(owner, "<span class = 'danger'>You feel confident enough to strike more and harder, and use this against toxins in your body.</span>")
+			attacks_w_flurry = 3
+			diamond_body = TRUE
+		if(12)
+			to_chat(owner, "<span class = 'danger'>You now use a circle kick to extend your attacks to other foes in combat.</span>")
+			circle_kick = TRUE
+		if(14)
+			to_chat(owner, "<span class = 'danger'>You use a downed foe to strike at their allies.</span>")
+			cleave_level = 2
+		if(15)
+			to_chat(owner, "<span class = 'danger'>You have practiced the Quivering Palm technique enough to use it in combat.</span>")
+			quivering_palm.Grant(owner)
+			quivering_palm.martial = src
+			available_actions = list(stunning_fist, wholeness_of_body, quivering_palm)

```

Comment:
update the list instead of making a brand new one, this one's my bad

Reply:
Done and done.